src/docs/arch-transaction.dox

```diff
@@ -5,4 +5,114 @@ A caller of WiredTiger uses @ref transactions within the API to start and stop t
 a session (thread of control).
 
 Internally, the current transaction state is represented by the WT_TXN structure.
+
+WiredTiger performs all the operations in transactions. If the user doesn't explicitly declare a transaction, WiredTiger will automatically create a transaction for the user's operation.
+
+@section Lifecycle
+
+A WiredTiger session creates and managers the transactions' whole lifecycle. A session is only able to manage one transaction at any given time and it cannot run another transaction until the current one completes. Since every session are executed on a single thread, all the operations in the transaction are executed sequentially on the same thread as well.
+
+@plantuml_start{transaction_lifecycle.png}
+@startuml{transaction_lifecycle.png}
+:Transaction Lifecycle;
+
+split
+    :perform a read operation
+    (Create an auto transaction);
+split again
+    :perform a write operation
+    (Create an auto transaction with a transaction id);
+split again
+    :declare the beginning of a transaction;
+    :perform read operations;
+    :perform a write operation
+    (Assign a transaction id);
+    :perform read write operations;
+split again
+    :declare the beginning of a transaction;
+    :perform read operations;
+    :perform a write operation
+    (Assign a transaction id);
+    :perform read write operations;
+    :prepare the transaction;
+end split
+
+split
+    :rollback;
+split again
+    :commit;
+end split
+
+Stop
+@enduml
+@plantuml_end
+
+A transaction starts when the user explicitly declares
```

Comment:
Whitespace on this line can be fixed.

Reply:
Probably worth being specific here and saying "A transaction starts in two scenarios, when the user calls begin via WT_SESSION::begin_transaction or internally when the user preforms and operation. Internally they are only started if they are not already within the context of a running transaction"

Reply:
"If declared explicitly the transaction will be active until..."

Reply:
Given that we talk about the transaction being started internally for operations a few times, we need to specify which operations either in the previous mention or here in my opinion.